### ISSUE TITLE
What is the state of QTodoTxt?

### ISSUE THREAD

**Issue by flight16** [QTodoTxt/QTodoTxt#155]
First, thanks for this project.  It's nice, tiny, and the syntax highlighting is useful.  

I noticed that the continuous integration build is marked as failed, there are 8 open pull requests, and the last commit was 1/2 year ago.  I want to add a few new features (sorting, saved filters, tweaks to behavior) and send pull requests.  Would my pull requests get integrated in a timely manner?

**Comment by mNantern** [comment]
Thank you !

Unfortunately I don't have enough time to work on QTT anymore :(

But I'm more than open to find a solution if someone want to work on the project !

**Comment by funkyfuture** [comment]
thanks by bringing this up.

i have also reduced capacities, but still am willing to maintain the code.

however, i haven't looked after this project for a while. the reason is, that i [refactored the internal task representation](https://github.com/funkyfuture/QTodoTxt/tree/lib-task) that is working according to the unittests, but there's some quirk that doesn't let a task's string be rendered in the ui. at some point i found it frustrating to merge in changes due to the effort it takes to rebase my branch upon these.

simply put: if we get the new task-interface working, i will revive maintaining.

another issue i'd like to address is that we agree on a roadmap and some few paradigms (e.g. don't implement features that are not in the reference implementation).

to procrastinate a little on another project, i'll go through issues and prs this afternoon.


ping @Asaika7 @oroulet @raabf @dkaparis @yddub @dittaeva

**Comment by kmic2006** [comment]
Hello! I stumbled over this project a few days ago and I'm interested in contributing. I reviewed a whole bunch of TODO-Software, and this one seems to fit best to my personal needs. Indeed, there are many things I would like trying to improve. If I compare the last release with the current master branch, the master branch is clearly better. There also seems to be many useful improvements in the pull requests.

I don't know anything about that refactoring of the internal task class, nor its importance, nor its goals. But I can see that it is not working for over one year now, and that it is blocking the introduction of new features. I don't think this is a desirable situation - work should go on.

Some words about me: I'm a C++ programmer, and I'm familiar with Qt and git. I'm complete new to Github and Python. Because I have exactly zero edits in github up to now, I would suggest that I first of all fork this repository and try to push my desired changed into it. After a while, you can evaluate my edits and maybe (@mNantern) we find then a solution. But I'm open to other suggestions too.

**Comment by oroulet** [comment]
Agree with bug owner here. It is not very reassuring when a project show no activities over a long time and the maintainer refuses to merge pull request because of bigger changes coming. It is a quite secure way to a project death.

In addition it seems the maintainer is against very simple changes like refusing contextual menu that would add a few lines of code while the current GUI code is completely over engineered with files of 100 lines that could have been 5 lines of code in main GUI file. 

Maybe it is time to fork project and create a github group for further  development.

On the other side the coming refactoring of task seems welcome...

**Comment by funkyfuture** [comment]
> I don't know anything about that refactoring of the internal task class, nor its importance, nor its goals.
Python is great for RAD. and like any piece of software that evolves over time, at some point there is the need to straighten things out or you will find yourself in hell when it comes to bugfixing or adding features.
the goal of that particular refactoring is to shift the handling of tasks from a simple parsing approach to a more abstract model that allows adding syntactic entities and a more flexible handling.
you may also have a look at the settings module. imo it's obvious that it needs to be implemented in a far more generalized way.

> After a while, you can evaluate my edits
i'll see that i review your PRs. unfortunately there's a long standing issue with PySide on the testing platform, which requires some extra effort.

> It is a quite secure way to a project death.
an unmaintainable code base is as well.

**Comment by bartgenuit** [comment]
Correct me if I'm wrong @funkyfuture, but isn't one of the features of using branches in developments like these that you can develop your refactoring (more or less) in parallel to other fixes? Only once the quirk has been fixed and the feature is ready to integrate do you have a need to rebase, right? Or am I missing something?
Then again, I understand that you care about this feature personally, and that getting it integrated will give quite a motivation boost. So maybe someone here could help you in fixing the refactoring, it sounds like some combined efforts of this community will make it work.

**Comment by mNantern** [comment]
I agree with both of you: we must improve the code base AND accept new PR. It's been too long since the last release of QTT.

I don't have enough time to code on it, but I still can make new release for everyone using QTT.

@funkyfuture do you have a little time to work on the refactoring with the community to make it work ?

@all can we agreed on a bug/PR list for the next release of QTT ? FYI a google group is open for QTT (https://groups.google.com/forum/#!forum/qtodotxt) if you want.

**Comment by oroulet** [comment]
@mNantern I propose to create a github group. that makes it very clear where the main repository is and easy for several people to work together. I already maintain a few projects so I can help with one more, but I am not neutral here, I want toolbar and contextual menu :-) I do most edit of todo file with vim. I only want an interface to have a nice overview and do small edit when I first started it, without having to learn any specific shortcuts

**Comment by kmicc** [comment]
A bug/PR list for the next release of QTT would be ok. But I don't think this is the most important thing to do right now. I think it would be more important to bring back this project into an actively maintained state. Therefore, I appreciate *any* change to the current situation, but I'm too new to github to be able to propose a solution or even to evaluate a proposed solution. Indeed, I prefer a solution *within* github.

If there is a consensus about a new project structure, I'm going do my best to contribute to it. If there will be no such consensus, I will continue working on my own fork. About the great goal: I'm going to make QTT more user friendly, not more minimalistic.

**Comment by mNantern** [comment]
@oroulet I like the idea of a github group. I just made that for QTT. Thank you !

**Comment by oroulet** [comment]
@mNantern great! now you need to add motivated people to the group, I haven't seen much happen and I found some new bugs today when setting new due date and month is december,,,, which should be fixed

**Comment by mNantern** [comment]
Any volunteer to be part of the group ? ping  @kmicc @oroulet @sdenel

**Comment by oroulet** [comment]
I would like to be part of the group.

**Comment by kmicc** [comment]
I would also like to join the group. (and +1 for some rules within the group)

**Comment by mNantern** [comment]
I added you both to QTT group and opened a new discussion in google groups to talk about rules before pushing data into master: https://groups.google.com/forum/#!topic/qtodotxt/cEseIN28Ieo

Thanks flight16 for bringing this up, it's been very useful ! I think we can close that issue and continue in Google groups !

**Comment by kmicc** [comment]
@mNantern Thank you! 

Unfortunately, I don't have a google account and I'm not too motivated to create one. Because I don't have any experience in managing collaborative projects I think it will be ok if I join the google group read-only.

@oroulet I read your proposed rules and think they are ok.

**Comment by oroulet** [comment]
OK I created a new issue called group rules so we can discuss things there about rules https://github.com/QTodoTxt/QTodoTxt/issues/169
